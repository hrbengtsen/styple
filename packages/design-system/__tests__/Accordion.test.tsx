import * as React from "react";
import {
  render,
  screen,
  fireEvent,
  RenderResult,
} from "@testing-library/react";
import {
  Accordion,
  AccordionItem,
  AccordionTrigger,
  AccordionContent,
} from "..";

describe("Test Accordion", () => {
  let rendered: RenderResult;

  beforeEach(() => {
    rendered = render(
      <Accordion type="single">
        <AccordionItem value="accordion-one">
          <AccordionTrigger>Accordion one</AccordionTrigger>
          <AccordionContent data-testid="content-one">
            A section of content one.
          </AccordionContent>
        </AccordionItem>
        <AccordionItem value="accordion-two">
          <AccordionTrigger>Accordion two</AccordionTrigger>
          <AccordionContent>A section of content two.</AccordionContent>
        </AccordionItem>
      </Accordion>
    );
  });

  it("should render as expected", () => {
    expect(rendered).toMatchSnapshot();
  });

  it("should be collapsible when type is single", () => {
    let trigger: HTMLElement = screen.getByText("Accordion one");
    let content: HTMLElement = screen.getByTestId("content-one");

    fireEvent.click(trigger);
    expect(content).toHaveAttribute("data-state", "open");
    fireEvent.click(trigger);
    expect(content).toHaveAttribute("data-state", "closed");
  });
});
