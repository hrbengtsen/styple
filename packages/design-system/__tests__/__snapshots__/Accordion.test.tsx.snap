// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test Accordion should render as expected 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="c-PJLV"
      >
        <div
          class="c-PJLV"
          data-state="closed"
        >
          <h3
            class="c-leHWbT"
            data-state="closed"
          >
            <button
              aria-controls="radix-:r1:"
              aria-expanded="false"
              class="c-brQuED"
              data-radix-collection-item=""
              data-state="closed"
              id="radix-:r0:"
              type="button"
            >
              Accordion one
              <svg
                class="lucide lucide-chevron-down"
                fill="none"
                height="24"
                stroke="currentColor"
                stroke-linecap="round"
                stroke-linejoin="round"
                stroke-width="2"
                viewBox="0 0 24 24"
                width="24"
                xmlns="http://www.w3.org/2000/svg"
              >
                <polyline
                  points="6 9 12 15 18 9"
                />
              </svg>
            </button>
            <div
              class="c-cWhwNz c-cWhwNz-dbVHzF-size-fill"
              data-orientation="horizontal"
              role="separator"
            />
          </h3>
          <div
            aria-labelledby="radix-:r0:"
            class="c-kDvOPF"
            data-state="closed"
            data-testid="content-one"
            hidden=""
            id="radix-:r1:"
            role="region"
            style="--radix-accordion-content-height: var(--radix-collapsible-content-height); --radix-accordion-content-width: var(--radix-collapsible-content-width);"
          />
        </div>
        <div
          class="c-PJLV"
          data-state="closed"
        >
          <h3
            class="c-leHWbT"
            data-state="closed"
          >
            <button
              aria-controls="radix-:r3:"
              aria-expanded="false"
              class="c-brQuED"
              data-radix-collection-item=""
              data-state="closed"
              id="radix-:r2:"
              type="button"
            >
              Accordion two
              <svg
                class="lucide lucide-chevron-down"
                fill="none"
                height="24"
                stroke="currentColor"
                stroke-linecap="round"
                stroke-linejoin="round"
                stroke-width="2"
                viewBox="0 0 24 24"
                width="24"
                xmlns="http://www.w3.org/2000/svg"
              >
                <polyline
                  points="6 9 12 15 18 9"
                />
              </svg>
            </button>
            <div
              class="c-cWhwNz c-cWhwNz-dbVHzF-size-fill"
              data-orientation="horizontal"
              role="separator"
            />
          </h3>
          <div
            aria-labelledby="radix-:r2:"
            class="c-kDvOPF"
            data-state="closed"
            hidden=""
            id="radix-:r3:"
            role="region"
            style="--radix-accordion-content-height: var(--radix-collapsible-content-height); --radix-accordion-content-width: var(--radix-collapsible-content-width);"
          />
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="c-PJLV"
    >
      <div
        class="c-PJLV"
        data-state="closed"
      >
        <h3
          class="c-leHWbT"
          data-state="closed"
        >
          <button
            aria-controls="radix-:r1:"
            aria-expanded="false"
            class="c-brQuED"
            data-radix-collection-item=""
            data-state="closed"
            id="radix-:r0:"
            type="button"
          >
            Accordion one
            <svg
              class="lucide lucide-chevron-down"
              fill="none"
              height="24"
              stroke="currentColor"
              stroke-linecap="round"
              stroke-linejoin="round"
              stroke-width="2"
              viewBox="0 0 24 24"
              width="24"
              xmlns="http://www.w3.org/2000/svg"
            >
              <polyline
                points="6 9 12 15 18 9"
              />
            </svg>
          </button>
          <div
            class="c-cWhwNz c-cWhwNz-dbVHzF-size-fill"
            data-orientation="horizontal"
            role="separator"
          />
        </h3>
        <div
          aria-labelledby="radix-:r0:"
          class="c-kDvOPF"
          data-state="closed"
          data-testid="content-one"
          hidden=""
          id="radix-:r1:"
          role="region"
          style="--radix-accordion-content-height: var(--radix-collapsible-content-height); --radix-accordion-content-width: var(--radix-collapsible-content-width);"
        />
      </div>
      <div
        class="c-PJLV"
        data-state="closed"
      >
        <h3
          class="c-leHWbT"
          data-state="closed"
        >
          <button
            aria-controls="radix-:r3:"
            aria-expanded="false"
            class="c-brQuED"
            data-radix-collection-item=""
            data-state="closed"
            id="radix-:r2:"
            type="button"
          >
            Accordion two
            <svg
              class="lucide lucide-chevron-down"
              fill="none"
              height="24"
              stroke="currentColor"
              stroke-linecap="round"
              stroke-linejoin="round"
              stroke-width="2"
              viewBox="0 0 24 24"
              width="24"
              xmlns="http://www.w3.org/2000/svg"
            >
              <polyline
                points="6 9 12 15 18 9"
              />
            </svg>
          </button>
          <div
            class="c-cWhwNz c-cWhwNz-dbVHzF-size-fill"
            data-orientation="horizontal"
            role="separator"
          />
        </h3>
        <div
          aria-labelledby="radix-:r2:"
          class="c-kDvOPF"
          data-state="closed"
          hidden=""
          id="radix-:r3:"
          role="region"
          style="--radix-accordion-content-height: var(--radix-collapsible-content-height); --radix-accordion-content-width: var(--radix-collapsible-content-width);"
        />
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
